apps/meteor/tests/end-to-end/api/livechat/04-dashboards.ts

```diff
@@ -1,19 +1,143 @@
+
+	const TOTAL_MESSAGES = {
+		min: 5,
+		max: 10,
+	};
+	const DELAY_BETWEEN_MESSAGES = {
+		min: 1000,
+		max: (inactivityTimeout - 1) * 1000,
+	};
+	const TOTAL_ROOMS = 7;
+
+	const simulateRealtimeConversation = async (chatInfo: IChatInfo[]) => {
+		const promises = chatInfo.map(async (info) => {
+			const { room, visitor } = info;
+
+			// send a few messages
+			const numberOfMessages = random(TOTAL_MESSAGES.min, TOTAL_MESSAGES.max);
+
+			// eslint-disable-next-line @typescript-eslint/no-unused-vars
+			for await (const _ of Array(numberOfMessages).keys()) {
+				// flip a coin to decide who will send the message
+				const willSendFromAgent = random(0, 1) === 1;
+
+				if (willSendFromAgent) {
+					await sendAgentMessage(room._id);
+				} else {
+					await sendMessage(room._id, faker.lorem.sentence(), visitor.token);
+				}
+
+				const delay = random(DELAY_BETWEEN_MESSAGES.min, DELAY_BETWEEN_MESSAGES.max);
+				await sleep(delay);
+			}
+
+			// Last message is always from visitor so that the chat doesn't get abandoned due to
+			// "Livechat_visitor_inactivity_timeout" setting
+			await sendMessage(room._id, faker.lorem.sentence(), visitor.token);
+		});
+
+		await Promise.all(promises);
+	};
+
+	before(async () => {
+		if (!IS_EE) {
+			return;
+		}
+
+		await updateSetting('Livechat_visitor_inactivity_timeout', inactivityTimeout);
+		await updateSetting('Livechat_enable_business_hours', false);
+
+		const chatInfo: IChatInfo[] = [];
+
+		// create dummy test data for further tests
+		const { department: createdDept, agent: agent1 } = await createDepartmentWithAnOnlineAgent();
+		department = createdDept;
+
+		const agent2 = await createAnOnlineAgent();
+		await addOrRemoveAgentFromDepartment(department._id, { agentId: agent2.user._id, username: agent2.user.username }, true);
+		agents.push(agent1);
+		agents.push(agent2);
+
+		// start a few chats
+		const { room: r1, visitor: v1 } = await startANewLivechatRoomAndTakeIt({ departmentId: department._id, agent: agent1.credentials });
+		const { room: r2, visitor: v2 } = await startANewLivechatRoomAndTakeIt({ departmentId: department._id, agent: agent1.credentials });
+		const { room: r3, visitor: v3 } = await startANewLivechatRoomAndTakeIt({ departmentId: department._id, agent: agent2.credentials });
+		const { room: r4, visitor: v4 } = await startANewLivechatRoomAndTakeIt({ departmentId: department._id, agent: agent2.credentials });
+		const { room: r5, visitor: v5 } = await startANewLivechatRoomAndTakeIt({ departmentId: department._id, agent: agent2.credentials });
+		const room5ChatStart = moment();
+		const { room: r6, visitor: v6 } = await startANewLivechatRoomAndTakeIt({ departmentId: department._id, agent: agent2.credentials });
+		const room6ChatStart = moment();
+		const { room: r7, visitor: v7 } = await startANewLivechatRoomAndTakeIt({ departmentId: department._id, agent: agent2.credentials });
+
+		chatInfo.push({ room: r1, visitor: v1, agent: agent1 });
```

Comment:
Instead of pushing, why not define `chatInfo` here with all the elements there already?